server/src/main/java/io/whitefox/services/DeltaSharesService.java

```diff
@@ -3,13 +3,17 @@
 import io.whitefox.api.deltasharing.model.Schema;
 import io.whitefox.api.deltasharing.model.Share;
 import io.whitefox.api.deltasharing.model.Table;
+import io.whitefox.persistence.memory.PTable;
 import java.util.List;
 import java.util.Optional;
 import java.util.concurrent.CompletionStage;
 
 public interface DeltaSharesService {
+
   CompletionStage<Optional<Share>> getShare(String share);
 
+  CompletionStage<Optional<PTable>> getTable(String share, String schema, String table);
```

Comment:
Service should either only return P classes or none. Since as of now we decided to not do it, we should move the logic that now is implemented in the api, to the service.
Moreover, "logic" should be part of the service, and the api should only adapt to the protocol (i.e. create response)